mh/time_utils.py

```diff
@@ -20,9 +20,44 @@
 """
 
 import datetime
+from math import floor
+
 
 EPOCH = datetime.datetime(1970, 1, 1)
+TICKS_PER_CYCLE = 3000
 
 
 def datetime_to_int(dt):
     return int((dt - EPOCH).total_seconds())
+
+JHEN_EVENT_OFFSET = 14  # Cycle of 14 days
+FOG_START = 0  
+JHEN_START = 1  # 1 day of fog
+JHEN_END = 3  # Followed by 2 days of sandstorm
+
+
+def current_tick():
+    return 4294967295 - datetime_to_int(datetime.datetime.now())
```

Comment:
Please use the hex notation (0xffffffff) to make it clearer and add some docstring to explain your rational. I have many doubts regarding it to be the right approach. Though, I'll let it slide for the time being since we don't have anything better.

Reply:
Switched to hex notation and added docstring explaining the current rationale